### PR TITLE
Catch check failures

### PR DESCRIPTION
**Pull request by kmclarnon**
This is mostly a port of the code that exists for our overwatch description listener, but this will write a separate file (that will also be picked up by overwatch) to indicate that there was a check failure.  We can then propagate this information to sfx and (most importantly) stop failing people's builds because one of our checks broke

@stevegutz @Xcelled

### CODE REVIEW

File: check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java [hunk 1]
```diff
@@ -902,13 +919,22 @@ public Void visitWildcard(WildcardTree tree, VisitorState visitorState) {
...
+  }
+
   /**
    * Handles an exception thrown by an individual BugPattern. By default, wraps the exception in an
    * {@link ErrorProneError} and rethrows. May be overridden by subclasses, for example to log the
    * error and continue.
    */
   @Override
-  protected void handleError(Suppressible s, Throwable t) {
+  protected void handleError(Suppressible s, Throwable t)  {
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java]: nit: ` `

File: check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java [hunk 2]
```diff
@@ -931,4 +957,84 @@ protected void handleError(Suppressible s, Throwable t) {
...
     return this.bugCheckers;
   }
+
+  private static void flushErrors() {
+    Optional<Path> output = getOutput();
+    if (!output.isPresent()) {
+      return;
+    }
+
+    try (OutputStream stream = new FileOutputStream(output.get().toFile())) {
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java]: You can do `Files.newOutputStream(output.get())` here

File: core/AutoValue_Test.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
+
```
Comment (stevegutz) [inline core/AutoValue_Test.java]: nit:empty lines. Should this have a package or is this following what google does?

File: check_api/pom.xml [hunk 4]
```diff
@@ -154,6 +154,12 @@
       <artifactId>caffeine</artifactId>
       <version>${caffeine.version}</version>
     </dependency>
+    <!-- Hubspot deps -->
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.7.9.5</version>
```
Comment (stevegutz) [inline check_api/pom.xml]: Should we use 2.9.9.x or 2.9.10.x here?

File: core/AutoValue_Test.java (hunk 3 above)
Comment (kmclarnon) [reply core/AutoValue_Test.java]: Hmm I didn't touch this file, I didn't notice it in the diff before. I'll figure out what happened here

File: check_api/pom.xml (hunk 4 above)
Comment (kmclarnon) [reply check_api/pom.xml]: This is going to be in the same classpath as several other things that are on basepom, so I'd like to keep the version consistent.

Comment (stevegutz) [reply check_api/pom.xml]: Ok. We should probably update basepom then (doesn't have to be now)

File: check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java [hunk 5]
```diff
@@ -137,12 +140,24 @@
 import com.sun.tools.javac.code.Symbol.CompletionFailure;
 import com.sun.tools.javac.util.JCDiagnostic.DiagnosticPosition;
 import com.sun.tools.javac.util.Name;
+
+import java.io.FileOutputStream;
```
Comment (stevegutz) [inline check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java]: nit: unused imports

Comment (kmclarnon) [reply check_api/src/main/java/com/google/errorprone/scanner/ErrorProneScanner.java]: :shakes-fist: intellij

**Pull request merged**